include/libsemigroups/knuth-bendix.hpp

```diff
@@ -115,20 +115,21 @@ namespace libsemigroups {
     // KnuthBendix - typedefs/aliases - private
     ////////////////////////////////////////////////////////////////////////
 
-    using external_string_type = std::string;
-    using internal_string_type = std::string;
-    using external_char_type   = char;
-    using internal_char_type   = char;
+    // using external_string_type = std::string;
+    // using internal_string_type = std::string;
+    // using external_char_type   = char;
+    // using internal_char_type   = char;
```

Comment:
These were defined both here and in `rewriters.hpp`. I've commented these out for now, and changed all occurrences to be the `detail` version (apart from in `KBE` because at a quick glance that looked like it was doing something different). Is this a reasonable thing to do?